### ISSUE TITLE
support for autocompleting package names, package versions, and features in Cargo.toml

### ISSUE THREAD

**Issue by kurtbuilds** [tamasfe/taplo#689]
There are a few efforts to create separate LSPs for autocompleting package names, package versions, and features in Cargo.toml (https://github.com/frederik-uni/cargotom; https://github.com/washanhanzi/cargo-appraiser/).

Is it in scope to add support for this feature to this project?

**Comment by panekj** [comment]
> Is it in scope to add support for this feature to this project?

Not really. Taplo is working solely off the JSON schemas as it's meant for TOML validation. Any support for specific file format such as e.g. Cargo should be handled separately.

**Comment by kurtbuilds** [comment]
Ok.

Right now taplo is the default LSP for Zed (my motivation), but it impedes first time UX to not fully support Cargo.toml, and it's redundant to maintain two DOM implementations, run two LSPs for the same file in parallel, etc.

I submitted a draft PR that starts to implement this, as I want this for my editor use, and I want to support that community. If it's out of scope, I'll discuss with Zed team what they prefer if taplo doesn't intend to fully support cargo.toml.
